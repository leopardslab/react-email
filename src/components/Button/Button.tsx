import { ReactNode, HTMLAttributeAnchorTarget } from 'react';
import type { BaseStyleProp } from '../types';
import { makeStyles } from '../../utils/makeStyles';
import { sx } from '../../utils/sx';
import { useTheme } from '../ThemeProvider';

export type ButtonStyles = 'root' | 'primary' | 'secondary';

/**
 * Interface for PropTypes for the `Button` component.
 */
export interface ButtonProps extends BaseStyleProp<ButtonStyles> {
  /**
   * The content of the component.
   */
  children?: ReactNode;
  /**
   * Applies the theme button styles.
   * @default 'primary'
   */
  variant?: 'primary' | 'secondary';
  /**
   * The URL to link to when the button is clicked.
   * An `a` element will be used as the root node.
   */
  href: string;
  /**
   * The target of the link.
   * @default '_blank'
   */
  target?: HTMLAttributeAnchorTarget;
}

const useStyles = makeStyles({
  root: {},
  primary: {},
  secondary: {},
});

export const Button = ({
  children,
  variant = 'primary',
  href,
  target = '_blank',
  classes,
  className,
}: ButtonProps): JSX.Element => {
  const themeClasses = useTheme('button');
  const styles = useStyles({ classes }, themeClasses);
  const buttonStyle = sx(styles.root, styles[variant]);

  return (
    <a href={href} target={target} style={buttonStyle} className={className}>
      {children}
    </a>
  );
};
