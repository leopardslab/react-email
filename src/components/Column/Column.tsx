import { CSSProperties, ReactNode } from 'react';
import { makeStyles } from '../../utils/makeStyles';
import { sx } from '../../utils/sx';
import type { BaseStyleProp } from '../types';
import { useTheme } from '../ThemeProvider';

export type ColumnStyles = 'root';

export interface ColumnProps extends BaseStyleProp<ColumnStyles> {
  children?: ReactNode;
  align?: 'left' | 'center' | 'right';
  verticalAlign?: CSSProperties['verticalAlign'];
}

const useStyles = makeStyles({
  root: {},
});

export const Column = ({
  children,
  className,
  classes,
  align = 'center',
  verticalAlign = 'top',
}: ColumnProps): JSX.Element => {
  const themeClasses = useTheme('column');
  const styles = useStyles({ classes }, themeClasses);

  return (
    <td className={className} style={sx(styles.root, { verticalAlign })} align={align}>
      {children}
    </td>
  );
};
