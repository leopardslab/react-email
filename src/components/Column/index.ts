export { Column, ColumnProps } from './Column';
