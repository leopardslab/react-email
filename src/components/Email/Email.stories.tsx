import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from './Email';

export default {
  component: Email,
} as ComponentMeta<typeof Email>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Email> = (args) => <Email {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: (
    <table style={{ width: '100%' }}>
      <tbody>
        <tr>
          <td>
            <p style={{ margin: '0', fontSize: '30px' }}>Hello World</p>
          </td>
        </tr>
      </tbody>
    </table>
  ),
  classes: {
    root: {
      backgroundColor: 'gray',
      color: 'white',
    },
  },
};
