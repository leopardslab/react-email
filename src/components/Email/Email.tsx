import { ReactNode } from 'react';
import { BaseStyleProp } from '../types';
import { makeStyles } from '../../utils/makeStyles';

type EmailStyles = 'root';

export interface EmailProps extends BaseStyleProp<EmailStyles> {
  children?: ReactNode;
}

const useStyles = makeStyles({
  root: {
    margin: '0px auto',
    maxWidth: '600px',
  },
});

export const Email = ({ children, className, classes }: EmailProps): JSX.Element => {
  const styles = useStyles({ classes });

  return (
    <div style={styles.root} className={className}>
      {children}
    </div>
  );
};
