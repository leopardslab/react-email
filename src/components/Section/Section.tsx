import { ReactNode } from 'react';

export interface SectionProps {
  children?: ReactNode;
  className?: string;
}

export const Section = ({ children, className }: SectionProps): JSX.Element => {
  return (
    <table
      cellPadding="0"
      cellSpacing="0"
      role="presentation"
      style={{ width: '100%', border: '0', verticalAlign: 'top' }}
      className={className}
    >
      <tr>{children}</tr>
    </table>
  );
};
