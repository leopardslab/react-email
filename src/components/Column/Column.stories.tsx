import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from '../Email/Email';
import { Column } from './Column';

export default {
  component: Column,
} as ComponentMeta<typeof Column>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Column> = (args) => (
  <Email>
    <Column {...args} />
  </Email>
);

export const Default = Template.bind({});

Default.args = {
  children: (
    <tr>
      <td style={{ fontSize: '30px', color: 'red' }}>Hello World</td>
    </tr>
  ),
};
