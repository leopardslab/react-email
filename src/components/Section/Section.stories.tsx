import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from '../Email/Email';
import { Column } from '../Column/Column';
import { Section } from './Section';

export default {
  component: Section,
} as ComponentMeta<typeof Section>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Section> = (args) => (
  <Email>
    <Column>
      <Section {...args} />
    </Column>
  </Email>
);

export const Default = Template.bind({});

Default.args = {
  children: <td style={{ fontSize: '30px', color: 'red' }}>This is a new section</td>,
};
