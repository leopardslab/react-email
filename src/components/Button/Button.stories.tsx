import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from '../Email/Email';
import { Section } from '../Section/Section';
import { Column } from '../Column/Column';
import { Button } from './Button';

export default {
  component: Button,
} as ComponentMeta<typeof Button>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Button> = (args) => (
  <Email>
    <Section>
      <Column>
        <Button {...args} />
      </Column>
    </Section>
  </Email>
);

export const Default = Template.bind({});

Default.args = {
  children: 'Click here',
  href: 'https://github.com/leopardslab/react-email',
  classes: {
    root: {
      backgroundColor: 'blue',
      color: 'white',
    },
  },
};
