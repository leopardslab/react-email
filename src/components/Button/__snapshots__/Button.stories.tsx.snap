// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <a href="https://github.com/leopardslab/react-email"
             target="_blank"
             style="font-size: 18px; text-decoration: none; padding: 10px 16px; border-radius: 5px; cursor: pointer; background-color: blue; color: white;"
          >
            Default Button
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/Button Primary smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <a href="https://github.com/leopardslab/react-email"
             target="_blank"
             style="font-size: 18px; text-decoration: none; padding: 10px 16px; border-radius: 5px; cursor: pointer; background-color: green; color: white;"
          >
            Primary Button
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/Button Secondary smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <a href="https://github.com/leopardslab/react-email"
             target="_blank"
             style="font-size: 18px; text-decoration: none; padding: 10px 16px; border-radius: 5px; cursor: pointer; background-color: red; color: white;"
          >
            Secondary Button
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
